### ISSUE TITLE
jpeg compression changes dtype

### ISSUE THREAD

**Issue by arsenyinfo** [albu/albumentations#32]
Current implementation of Jpeg compression works only with uint8 data, so it can break some pipelines where data is casted to `[0, 1]` floats from the very beginning. 

```
In [29]: from albumentations.augmentations import functional as f_aug

In [30]: img = np.random.rand(100, 100).astype('float32')

In [31]: f_aug.jpeg_compression(img, 95).dtype
Out[31]: dtype('uint8')
```

Please let me know if you'd like to get PR on this issue.

P.S. BTW, it may be a good idea to check for input and output dtype equality to make the debugging of similar issues easier.

**Comment by creafz** [comment]
A pull request that adds support for float inputs will be greatly appreciated. 

I checked `imencode` and it looks like it converts all non-uint8 inputs to uint8 [[source]](https://github.com/opencv/opencv/blob/master/modules/imgcodecs/src/loadsave.cpp#L912-L918).

I think the following steps could be used to work with float inputs in `jpeg_compression`:
1. Multiply the input by 255 and cast it to uint8.
2. Apply cv2.imencode and cv2.imdecode functions as before.
3. Cast the output of cv2.imdecode back to float, divide all values by 255 and return the resulting image.

However, I could be mistaken and maybe there is a better way of handling float inputs.

As for checking outputs and their data types, we have some tests in [tests/test_augmentations.py](https://github.com/albu/albumentations/blob/master/tests/test_augmentations.py) that verify output values of augmentations. `jpeg_compression` is not tested with float inputs because in the latest version it should raise an exception if the data type of the input does not equal np.uint8 https://github.com/albu/albumentations/blob/master/albumentations/augmentations/functional.py#L216

**Comment by creafz** [comment]
Fixed in #38
